File: 70. Climbing Stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        a = [1,1]
        for i in range(n-1):
            a.append(a[-1]+a[-2])
        return a[n]
        
-------------------------------------------------------
class Solution:
    def climbStairs(self, n: int) -> int:
        if (n<2):
            return n
        count=[1,2]
        for i in range(2,n):
            count.append(count[i-1]+count[i-2])
        return count[n-1]
--------------------------------------------------------

File: 70. Climbing Stairs.java
class Solution {
    public int climbStairs(int n) {
        int dp[]=new int [n+1];
        dp[0]=1;
        dp[1]=1;
        for(int i=2;i<=n;i++) {
            dp[i]=dp[i-1] + dp[i-2];
        }
        return dp[n];
    }
}
-----------------------------------------------------
  
class Solution {
    public int climbStairs(int n) { //FIBONACCI
        if(n<=2)
            return n;
        int one =1;
        int two =2;
        for(int i = 3;i<=n;i++){
            int temp = one;
            one = two;
            two+=temp;
        }
        return two;
    }
}
